draft/mson-parameters-headers.md

```diff
@@ -0,0 +1,216 @@
+---
+RFC: XXXX
+Author: Z
+Status: Draft
+Created: 2015-09-23
+Last Modified: 2015-09-28
+---
+
+# API Blueprint RFC XXXX: MSON Parameters and Headers
+
+## Table of Contents
+
+- [Abstract](#abstract)
+- [Motivation](#motivation)
+- [Implementation](#implementation)
+- [Backwards Compatibility](#backwards-compatibility)
+
+## Abstract
+
+This RFC proposes using full [MSON][] syntax for description of URI parameters
+and HTTP headers.
+
+## Motivation
+
+Using MSON syntax for description of URI parameters and HTTP Headers will
+greatly help to unify the API Blueprint syntax and introduce reusability of
+URI parameters and HTTP headers. It will also greatly introduce the reusability
+of other data structures.
+
+In addition it will address many issues previously tracked on API Blueprint
+GitHub repository.
+
+### URI Parameters Issues Tracked
+
+This RFC addresses following issues:
+
+1. [Usage of data structures in Parameters](https://github.com/apiaryio/api-blueprint/issues/250)
+1. [Global query URI parameter](https://github.com/apiaryio/api-blueprint/issues/24)
+1. [Description of enumeration elements](https://github.com/apiaryio/api-blueprint/issues/119)
+1. [Array in parameters](https://github.com/apiaryio/api-blueprint/issues/240)
+1. [Minimum/Maximum values for URI Parameters](https://github.com/apiaryio/api-blueprint/issues/211)
+
+    NOTE: This is going to be addressed once MSON introduces validation.
+
+### HTTP Headers Issues Tracked
+
+This RFC addresses following issues:
+
+1. [Parametric Headers Description](https://github.com/apiaryio/api-blueprint/issues/26)
+1. [Referenced HTTP headers](https://github.com/apiaryio/api-blueprint/issues/17)
+1. [Support for multiline syntax header](https://github.com/apiaryio/api-blueprint/issues/60)
+
+In addition this issue is solved by this RFC:
+
+1. [Reuse headers and attributes](https://github.com/apiaryio/api-blueprint/issues/259)
+
+## Implementation
+
+### Parameters
+
+The implementation should assume the existing syntax for API Blueprint
+[URI parameters section][] is, in fact, an [MSON type declaration][].
+
+The type being declared inherits from a *Parameters base type* unless specified
+otherwise. If another base type is specified it must inherit from the
+*Parameters base type*.
```

Comment:
Bear in mind I'm not very deeply knowledgeable (or experienced) in API Blueprint. That being said, It seems that requesting to inherit from a base parameter type means some for of multiple inheritance feature. It would introduce some possible type safety but in that case shouldn't there also be a Response base type and so on?

Given that it should already possible for users to create a Parameter base type if they want to, then maybe it's best to lower the barrier to entry and allow any arbitrary MSON data structure as @pksunkara suggests.